### ISSUE TITLE
Incorrect rsync excution in Pods-resources.sh

### ISSUE THREAD

**Issue by lucaseverini** [CocoaPods/CocoaPods#4172]
Hi everyone,
This issue seems quite similar to couple other already listed, however they aren't exactly the same so
I post it anyway. I apologize in advance if I'm just adding some noise...

On the latest (GM) version of Xcode 7 trying to build my iOS 8 app, which build just fine with Xcode 6, I gets this error:
`Command /bin/sh failed with exit code 23`
Which is caused by incorrect arguments (paths) passed to rsync as can be seen in the log.
The first part of those path below is duplicated and therefore the path is incorrect.

```
sending incremental file list
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPArrowLeft.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPArrowLeft.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPArrowRight.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPArrowRight.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPCheckmark.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPCheckmark.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPCloseBtn.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPCloseBtn.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPDismissKeyboard.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPDismissKeyboard.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPLogo.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPLogo.imageset" failed: No such file or directory (2)
rsync: change_dir "/Jobs/Sommly/sommly-frontend/Jobs/Sommly/sommly-frontend/Pods/SVProgressHUD/SVProgressHUD" failed: No such file or directory (2)
sent 18 bytes  received 12 bytes  60.00 bytes/sec
total size is 0  speedup is 0.00
rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1040) [sender=3.0.4]
```

Here is the complete output of that build phase "Copy Pods resources":
```
PhaseScriptExecution Copy\ Pods\ Resources Build/Intermediates/Sommly.build/Debug-iphoneos/Sommly\ Beta.build/Script-815676381B2E5F3900497142.sh
    cd /Jobs/Sommly/sommly-frontend
    /bin/sh -c \"/Jobs/Sommly/sommly-frontend/Build/Intermediates/Sommly.build/Debug-iphoneos/Sommly\ Beta.build/Script-815676381B2E5F3900497142.sh\"

/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPArrowLeft.imageset/MPArrowLeft.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPArrowLeft.imageset/MPArrowLeft@2x.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPArrowRight.imageset/MPArrowRight.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPArrowRight.imageset/MPArrowRight@2x.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPCheckmark.imageset/MPCheckmark.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPCheckmark.imageset/MPCheckmark@2x.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPCloseBtn.imageset/MPCloseBtn.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPCloseBtn.imageset/MPCloseBtn@2x.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPDismissKeyboard.imageset/MPDismissKeyboard.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPDismissKeyboard.imageset/MPDismissKeyboard@2x.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPLogo.imageset/MPLogo.png
/Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/Media.xcassets/MPLogo.imageset/MPLogo@2x.png
ibtool --reference-external-strings-file --errors --warnings --notices --output-format human-readable-text --compile /Jobs/Sommly/sommly-frontend/Build/Products/Debug-iphoneos/Sommly Beta.app/MPNotification.storyboardc /Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/MPNotification.storyboard --sdk /Developer/Xcode-6.3.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS8.3.sdk
ibtool --reference-external-strings-file --errors --warnings --notices --output-format human-readable-text --compile /Jobs/Sommly/sommly-frontend/Build/Products/Debug-iphoneos/Sommly Beta.app/MPSurvey.storyboardc /Jobs/Sommly/sommly-frontend/Pods/Mixpanel/Mixpanel/MPSurvey.storyboard --sdk /Developer/Xcode-6.3.1.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS8.3.sdk
/Jobs/Sommly/sommly-frontend/Pods/SVProgressHUD/SVProgressHUD/SVProgressHUD.bundle
####### PODS_ROOT:  /Jobs/Sommly/sommly-frontend/Pods
####### INSTALL_DIR:  /tmp/Sommly.dst/Applications
####### RESOURCES_TO_COPY:  /Jobs/Sommly/sommly-frontend/Pods/resources-to-copy-Sommly Beta.txt
####### CONFIGURATION_BUILD_DIR:  /Jobs/Sommly/sommly-frontend/Build/Products/Debug-iphoneos
####### UNLOCALIZED_RESOURCES_FOLDER_PATH:  Sommly Beta.app
building file list ... done
MPArrowLeft.png
MPArrowLeft@2x.png
MPArrowRight.png
MPArrowRight@2x.png
MPCheckmark.png
MPCheckmark@2x.png
MPCloseBtn.png
MPCloseBtn@2x.png
MPDismissKeyboard.png
MPDismissKeyboard@2x.png
MPLogo.png
MPLogo@2x.png
SVProgressHUD.bundle/
SVProgressHUD.bundle/angle-mask.png
SVProgressHUD.bundle/angle-mask@2x.png
SVProgressHUD.bundle/angle-mask@3x.png
SVProgressHUD.bundle/error.png
SVProgressHUD.bundle/error@2x.png
SVProgressHUD.bundle/error@3x.png
SVProgressHUD.bundle/info.png
SVProgressHUD.bundle/info@2x.png
SVProgressHUD.bundle/info@3x.png
SVProgressHUD.bundle/success.png
SVProgressHUD.bundle/success@2x.png
SVProgressHUD.bundle/success@3x.png

sent 53124 bytes  received 554 bytes  107356.00 bytes/sec
total size is 51533  speedup is 0.96
```

Best regards,

Luca Severini

**Comment by mboe** [comment]
Same here. Actually this is a blocker. Any workaround suggestion?

**Comment by lucaseverini** [comment]
/Pods/Target Support Files/Pods/Pods-resources.sh

**Comment by lucaseverini** [comment]
I think the issue is just in .../Pods/Target Support Files/Pods/Pods-resources.sh and someone who knows shell scripting well should be able to (easily) fix it... ;-)

**Comment by lucaseverini** [comment]
Hi everyone,
I've found a workaround to build the project with xcode 7 and makes me think the root of the problem may be not in cocoapods or, at least, not only there.
I noticed that xcode 6.4 that builds the project correctly the Pods-resources.sh script uses the rsync at /usr/bin whose version "is 2.6.9  protocol version 29".
Xcode 7 instead uses the rsync in /usr/local/bin whose version is "version 3.0.4  protocol version 30".
Renaming the rsync in /usr/local/bin to something else foced the use of the "good" rsync in /usr/bin and the project was built correctly. Typical... ;-)
Hope this helps.
